path_config.go

```diff
@@ -160,12 +180,32 @@ func (b *jwtAuthBackend) pathConfigWrite(ctx context.Context, req *logical.Reque
 	case config.OIDCDiscoveryURL != "":
 		_, err := b.createProvider(config)
 		if err != nil {
-			return logical.ErrorResponse(errwrap.Wrapf("error checking discovery URL: {{err}}", err).Error()), nil
+			return logical.ErrorResponse(errwrap.Wrapf("error checking oidc discovery URL: {{err}}", err).Error()), nil
 		}
 
 	case config.OIDCClientID != "" && config.OIDCDiscoveryURL == "":
 		return logical.ErrorResponse("'oidc_discovery_url' must be set for OIDC"), nil
 
+	case config.JWKSURL != "":
+		ctx, err := b.createCAContext(context.Background(), config.JWKSCAPEM)
+		if err != nil {
+			return logical.ErrorResponse(errwrap.Wrapf("error checking jwks_ca_pem: {{err}}", err).Error()), nil
+		}
+
+		keyset := oidc.NewRemoteKeySet(ctx, config.JWKSURL)
+
+		// Try to verify a correctly formatted JWT. The signature will fail to match, but other
+		// errors with fetching the remote keyset should be reported.
+		testJWT := "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.e30.Hf3E3iCHzqC5QIQ0nCqS1kw78IiQTRVzsLTuKoDIpdk"
+		_, err = keyset.VerifySignature(ctx, testJWT)
+		if err == nil {
+			err = errors.New("unexpected verification of JWT")
+		}
+
+		if !strings.Contains(err.Error(), "failed to verify id token signature") {
```

Comment:
Question, where does this error string come from? Is it coming from our code or some remote API of the user's somewhere?

Reply:
It is from the library we use. I don't love this approach, but it's not too bad given that it's not an API response. If they change this message, our tests will fail which is 👍 .